app/models/notification.rb

```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```

Comment:
Really, I don't understand what you are waiting for me. You want me to run these tests on my computer ?